tests/sample_size/scripts/test_utils.py

```diff
@@ -0,0 +1,269 @@
+        test_input_float = -0.1
+        mock_input.return_value = "n"
+        mock_get_float.return_value = test_input_float
+
+        with self.assertRaises(Exception) as context:
+            utils.get_alpha()
+
+        self.assertEqual(
+            context.exception.args[0],
+            "Error: Please provide a float between 0 and 0.3 for alpha.",
+        )
+
+    @patch("sample_size.scripts.utils.get_float")
+    @patch("sample_size.scripts.utils.input")
+    def test_get_mde(self, mock_input, mock_get_float):
+        test_metric_type = "boolean"
+        test_mde = 0.01
+        mock_input.return_value = test_mde
+        mock_get_float.return_value = test_mde
+
+        mde = utils.get_mde(test_metric_type)
+
+        self.assertEqual(mde, test_mde)
+        mock_input.called_once_with(
+            f"Enter the absolute minimum detectable effect for this {test_metric_type} \n"
+            f"MDE: targeted treatment metric value minus the baseline value"
+        )
+        mock_get_float.called_once_with(test_mde, "minimum detectable effect")
+
+    @patch("sample_size.scripts.utils.input")
+    def test_get_variable_from_input_boolean(self, mock_input):
+        mock_input.return_value = " Boolean "
+
+        metric_type = utils.get_metric_type_from_input()
+
+        self.assertEqual(metric_type, "boolean")
+
+    @patch("sample_size.scripts.utils.input")
+    def test_get_variable_from_input_numeric(self, mock_input):
+        mock_input.return_value = " Numeric "
+
+        metric_type = utils.get_metric_type_from_input()
+
+        self.assertEqual(metric_type, "numeric")
+
+    @patch("sample_size.scripts.utils.input")
+    def test_get_variable_from_input_ratio(self, mock_input):
+        mock_input.return_value = " Ratio "
+
+        metric_type = utils.get_metric_type_from_input()
+
+        self.assertEqual(metric_type, "ratio")
+
+    @patch("sample_size.scripts.utils.input")
+    def test_get_variable_from_input_error(self, mock_input):
+        mock_input.return_value = "test"
+
+        with self.assertRaises(Exception) as context:
+            utils.get_metric_type_from_input()
+
+        self.assertEqual(
+            context.exception.args[0],
+            "Error: Unexpected metric type. Please enter Boolean, Numeric, or Ratio.",
+        )
+
+    @patch("sample_size.scripts.utils.get_float")
+    @patch("sample_size.scripts.utils.input")
+    def test_get_variable_parameters(self, mock_input, mock_get_float):
+        test_input_float = 5
+        mock_input.return_value = test_input_float
+        mock_get_float.return_value = test_input_float
+        test_parameter_definitions = {
+            "test": "test test",
+            "case": "case case",
+        }
+
+        result = utils.get_variable_parameters(test_parameter_definitions)
+
+        self.assertEqual(mock_input.call_count, 2)
+        assert_equal(mock_input.call_args_list[0][0][0], "Enter the test test: ")
```

Comment:
why to use assert_equal here? self.assertEqual should just work?
```suggestion
        self.assertEqual(mock_input.call_args_list[0][0][0], "Enter the test test: ")
```